### PR TITLE
Command bar docs

### PR DESCRIPTION
**Pull request by Virrum**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1157258861200514)

### CODE REVIEW

File: Source/Controls/CommandBar/for_CommandBar/when_initiating_a_command_bar.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    commandBar = new CommandBar();
+  });
+
+  it('should have commandLocation set to "right"', () => commandBar.commandLocation.should.equal(commandLocation.right));
+  it('should have commandLabelPostition set to "right"', () => commandBar.commandLabelPosition.should.equal(commandLabelPosition.right));
+  it('should have zIndex set to 99', () => commandBar.zIndex.should.equal(99));
+  it('should not display secondary commands', () => commandBar.displaySecondaryCommands.should.be.false);
+});
+
+describe('when initiating a command_bar with commandLocation set', () => {
```
Comment (einari) [inline Source/Controls/CommandBar/for_CommandBar/when_initiating_a_command_bar.ts]: You should split these into multiple files and instead have a `when_initiating` folder with the different specifications as files underneath it. I would also use initializing or creating as the verb and possibly leave out *command_bar* in the spec name, as it is implicitly sitting in the context of the command_bar. 

Have a look [here](https://github.com/dolittle-interaction/JavaScript.Client/tree/master/Source/commands/for_CommandCoordinator) for an example of how we tend to do things. Unfortunately we have yet to document our current expectations around structure and how specs are to be structured and written.

But a rule of thumb is if you have multiple things within a «verb» (action), then create a folder and have the files start with the state you’re specifying or even start with an «and_» or similar. For your scenario the files would be «without_any_properties_set.ts», «with_command_label_position_set».

File: Source/Controls/CommandBar/CommandBarButton.ts [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
 @containerless()
 export class CommandBarButton {
-    @bindable icon: any;
-    @bindable label: any;
-    @bindable linkUrl: any;
-    constructor(){
-
-    }
+  @bindable label: String = '';
+  @bindable linkUrl: String = '';
```
Comment (einari) [inline Source/Controls/CommandBar/CommandBarButton.ts]: This is very limiting to specific behavior of navigating somewhere and something the *CommandBarButton* shouldn’t really have the responsibility for. This should be decided on by the consuming view+ViewModel. So typically a click event would be what we’d expect and then let whoever is consuming and using this decide wether or not its a navigational action or something else.

**Review by einari — CHANGES_REQUESTED** [review]
Love seeing this coming together. Great work!
Have a couple of comments here. The responsibilities should be very clear in all components and typically something like navigation shouldn’t be something we put onto the component.

**Comment by einari** [comment]
Now that I went back to our board, I see you have a task for a callback function.
Question is; does it make sense to have `linkUrl` there or not as an option. What you typically would need then is also the ability to specify target for it to be a fully viable solution. 

I'm slightly conflicted, but I feel there is something to be said for moving that responsibility to the consumer of the control instead.

File: Source/Controls/CommandBar/CommandBarButton.ts (hunk 2 above)
Comment (Virrum) [reply Source/Controls/CommandBar/CommandBarButton.ts]: So. this component has been hanging around for a while, and was made for a different purpose at the time that it actually should be aimed for. Perhaps we should remove linkUrl as you say and add the callback function right away. Why I created a task for it was mostly that this branch has been running for a couple of weeks now, and wanted to get this done and then add the correct behavior afterwards. But I can do it now.

How about a different button; CommandBarLinkButton ?

**Review by einari — APPROVED** [review]

**Review by einari — APPROVED** [review]

**Pull request merged**